<launch>
    <arg name="driving_line_topic" default="/driving_line"/>
    <arg name="uavcan_topic" default="/canros/msg/phoenix_msgs/NucDriveCommand"/>
    <arg name="trajectory_meta_input_topic" default="/trajectory_meta_commands"/>
    <arg name="vision_debug" default="true"/>

    <arg name="camera" default="camera"/>
    <arg name="homography_topic" default="/$(arg camera)/camera_homography"/>
    <arg name="camera_info_topic" default="/$(arg camera)/camera_info"/>

    <arg name="debug_image_in" default="/line_detection/debug_image"/>
    <arg name="debug_image_out" default="/trajectory_controller/debug_image"/>
    <arg name="parking_in_progress_service" default="/parking_in_progress"/>

    <arg name="output" default="screen"/>

    <group if="$(arg vision_debug)">
        <node name="trajectory_line_creator_node" pkg="drive_ros_trajectory_generator" type="trajectory_line_creator_node" respawn="false" output="$(arg output)">
            <remap from="line_in" to="$(arg driving_line_topic)"/>
            <remap from="can_topic" to="$(arg uavcan_topic)"/>
            <remap from="meta_in" to="$(arg trajectory_meta_input_topic)"/>
            <remap from="debug_image_in" to="$(arg debug_image_in)"/>
            <remap from="debug_image_out" to="$(arg debug_image_out)"/>
            <rosparam command="load" file="$(find drive_ros_image_recognition)/config/homography.yaml"/>
            <remap from="homography_in" to="$(arg homography_topic)"/>
            <remap from="camera_info" to="$(arg camera_info_topic)"/>
            <remap from="parallel_parking" to="/canros/msg/phoenix_msgs/ParallelParking"/>
            <remap from="parking_in_progress" to="$(arg parking_in_progress_service)"/>
            <rosparam command="load" file="$(find drive_ros_trajectory_generator)/config/car_params.yaml"/>
        </node>
    </group>

    <group unless="$(arg vision_debug)">
        <node name="trajectory_line_creator_node" pkg="drive_ros_trajectory_generator" type="trajectory_line_creator_node" respawn="false" output="$(arg output)">
            <remap from="line_in" to="$(arg driving_line_topic)"/>
            <remap from="can_topic" to="$(arg uavcan_topic)"/>
            <remap from="meta_in" to="$(arg trajectory_meta_input_topic)"/>
            <remap from="parallel_parking" to="/canros/msg/phoenix_msgs/ParallelParking"/>
            <remap from="parking_in_progress" to="$(arg parking_in_progress_service)"/>
            <rosparam command="load" file="$(find drive_ros_trajectory_generator)/config/car_params.yaml"/>
        </node>
    </group>
</launch>
